src/SDL/Input/GameController.hs

```diff
@@ -1,3 +1,81 @@
+{-# LANGUAGE DeriveDataTypeable #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+
 module SDL.Input.GameController
-  (
+  ( ControllerButton(..)
+  , ControllerButtonState(..)
+  , ControllerDeviceConnection(..)
   ) where
+
+import Data.Data (Data)
+import Data.Typeable
+import Data.Word
+import GHC.Generics (Generic)
+import GHC.Int (Int32)
+import SDL.Internal.Numbered
+import qualified SDL.Raw as Raw
+
+-- | Identifies a gamepad button.
+data ControllerButton
+  = ControllerButtonInvalid
+  | ControllerButtonA
+  | ControllerButtonB
+  | ControllerButtonX
+  | ControllerButtonY
+  | ControllerButtonBack
+  | ControllerButtonGuide
+  | ControllerButtonStart
+  | ControllerButtonLeftStick
+  | ControllerButtonRightStick
+  | ControllerButtonLeftShoulder
+  | ControllerButtonRightShoulder
+  | ControllerButtonDpadUp
+  | ControllerButtonDpadDown
+  | ControllerButtonDpadLeft
+  | ControllerButtonDpadRight
+  | ControllerButtonMax
```

Comment:
`SDL_CONTROLLER_BUTTON_MAX` isn't an actual button. SDL2 uses this enum value internally to size arrays for various purposes.

Will you consider removing `ControllerButtonMax` and map `SDL_CONTROLLER_BUTTON_MAX` to `ControllerButtonInvalid` in `FromNumber` instead?